core/src/main/java/org/apache/carbondata/core/util/TrashUtil.java

```diff
@@ -0,0 +1,224 @@
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.carbondata.common.logging.LogServiceFactory;
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.datastore.filesystem.CarbonFile;
+import org.apache.carbondata.core.datastore.impl.FileFactory;
+import org.apache.carbondata.core.mutate.CarbonUpdateUtil;
+import org.apache.carbondata.core.util.path.CarbonTablePath;
+
+import org.apache.hadoop.io.IOUtils;
+import org.apache.log4j.Logger;
+
+/**
+ * Mantains the trash folder in carbondata. This class has methods to copy data to the trash and
+ * remove data from the trash.
+ */
+public final class TrashUtil {
+
+  private static final Logger LOGGER =
+      LogServiceFactory.getLogService(TrashUtil.class.getName());
+
+  /**
+   * Base method to copy the data to the trash folder.
+   *
+   * @param sourcePath      the path from which to copy the file
+   * @param destinationPath the path where the file will be copied
+   * @return
+   */
+  private static void copyToTrashFolder(String sourcePath, String destinationPath)
+    throws IOException {
+    DataOutputStream dataOutputStream = null;
+    DataInputStream dataInputStream = null;
+    try {
+      dataOutputStream = FileFactory.getDataOutputStream(destinationPath);
+      dataInputStream = FileFactory.getDataInputStream(sourcePath);
+      IOUtils.copyBytes(dataInputStream, dataOutputStream, CarbonCommonConstants.BYTEBUFFER_SIZE);
+    } catch (IOException exception) {
+      LOGGER.error("Unable to copy " + sourcePath + " to the trash folder", exception);
+      throw exception;
+    } finally {
+      CarbonUtil.closeStreams(dataInputStream, dataOutputStream);
+    }
+  }
+
+  /**
+   * The below method copies the complete a file to the trash folder.
+   *
+   * @param filePathToCopy           the files which are to be moved to the trash folder
+   * @param trashFolderWithTimestamp timestamp, partition folder(if any) and segment number
+   * @return
+   */
+  public static void copyFileToTrashFolder(String filePathToCopy,
+      String trashFolderWithTimestamp) throws IOException {
+    CarbonFile carbonFileToCopy = FileFactory.getCarbonFile(filePathToCopy);
+    try {
+      String destinationPath = trashFolderWithTimestamp + CarbonCommonConstants
+          .FILE_SEPARATOR + carbonFileToCopy.getName();
+      if (!FileFactory.isFileExist(destinationPath)) {
+        copyToTrashFolder(filePathToCopy, destinationPath);
+      }
+    } catch (IOException e) {
+      // in case there is any issue while copying the file to the trash folder, we need to delete
+      // the complete segment folder from the trash folder. The trashFolderWithTimestamp contains
+      // the segment folder too. Delete the folder as it is.
+      FileFactory.deleteFile(trashFolderWithTimestamp);
+      LOGGER.error("Error while creating trash folder or copying data to the trash folder", e);
```

Comment:
in this long better to print to path also, it will be helpful for any issues in future

Reply:
done